src/coloring/acyclic_coloring.jl

```diff
@@ -0,0 +1,209 @@
+    
+    edge = find_edge(g, w, x)
+    e = find_root(set, edge)
+    p, q = firstVisitToTree[edge_index(g, e)]
+    if p != v
+        firstVisitToTree[edge_index(g, e)] = (v, w)
+    elseif q != w
+        forbiddenColors[color[x]] = v
+    end
+end
+
+"""
+        min_index(forbiddenColors::AbstractVector{<:Integer}, v::Integer)
+
+Returns min{i > 0 such that forbiddenColors[i] != v}
+"""             
+function min_index(forbiddenColors::AbstractVector{<:Integer}, v::Integer)
+    for i = 1:length(forbiddenColors)
+        if forbiddenColors[i] != v
+            return i
+        end
+    end
+end
+
+"""
+        grow_star(v::Integer,
+                w::Integer,
+                g::LightGraphs.AbstractGraph
+                firstNeighbor::Array{Tuple{Integer, Integer}, 1},
+                set::DisjointSets{LightGraphs.Edge})
+
+Subroutine to grow a 2-colored star after assigning a new color to the 
+previously uncolored vertex v, by comparing it with the adjacent vertex w.
+Disjoint set is used to store stars in sets, which are identified through key
+edges present in g. 
+"""
+function grow_star(v::Integer,
+                   w::Integer,
+                   g::LightGraphs.AbstractGraph,
+                   firstNeighbor::AbstractArray{<: Tuple{Integer, Integer}, 1},
+                   set::DisjointSets{LightGraphs.Edge},
+                   color::AbstractVector{<: Integer})
+    edge = find_edge(g, v, w)
+    push!(set, edge)
+    p, q = firstNeighbor[color[w]]
+    if p != v
+        firstNeighbor[color[w]] = (v, w)
+    else
+        edge1 = find_edge(g, v, w)
+        edge2 = find_edge(g, p, q)
+        e1 = find_root(set, edge1)
+        e2 = find_root(set, edge2)
+        union!(set, e1, e2)
+    end
+end
+
+
+"""
+        merge_trees(v::Integer,
+                w::Integer, 
+                x::Integer,
+                g::LightGraphs.AbstractGraph,
+                set::DisjointSets{LightGraphs.Edge})
+
+Subroutine to merge trees present in the disjoint set which have a
+common edge.
+"""
+function mergeTrees(v::Integer,
+                    w::Integer, 
+                    x::Integer,
+                    g::LightGraphs.AbstractGraph,
+                    set::DisjointSets{LightGraphs.Edge})
+    edge1 = find_edge(g, v, w)
+    edge2 = find_edge(g, w, x)
+    e1 = find_root(set, edge1)
+    e2 = find_root(set, edge2)
+    if (e1 != e2)
+        union!(set, e1, e2)
+    end
+end
+
+
+"""
+        find_edge(g::LightGraphs.AbstractGraph, v::Integer, w::Integer)
+
+Returns an edge object of the type LightGraphs.Edge which represents the 
+edge connecting vertices v and w of the undirected graph g
+"""
+function find_edge(g::LightGraphs.AbstractGraph, 
+                   v::Integer, 
+                   w::Integer)
+    for e in edges(g)
+        if (src(e) == v && dst(e) == w) || (src(e) == w && dst(e) == v)
+            return e
+        end
+    end
+    throw(error("$v and $w are not connected in graph g"))
```

Comment:
```suggestion
    throw(ArgumentError("$v and $w are not connected in graph g"))
```
It's better to use specific errors when possible